### PR TITLE
add periodic job UUID to upstream job id and use execution log as the…

### PR DESCRIPTION
**Pull request by wy65701436**
… periodic log

Signed-off-by: wang yan <wangyan@vmware.com>

### CODE REVIEW

File: src/common/job/client.go [hunk 1]
```diff
@@ -103,6 +105,36 @@ func (d *DefaultClient) GetJobLog(uuid string) ([]byte, error) {
 	return data, nil
 }
 
+// GetExecutions ...
+func (d *DefaultClient) GetExecutions(periodicJobID string) ([]job.Stats, error) {
+	url := d.endpoint + "/api/v1/jobs/" + periodicJobID + "/executions"
```
Comment (steven-zou) [inline src/common/job/client.go]: how about trying 'fmt.Sprintf("%s/%s/%s",1,1,1)'

File: src/common/job/client.go [hunk 2]
```diff
@@ -103,6 +105,36 @@ func (d *DefaultClient) GetJobLog(uuid string) ([]byte, error) {
...
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := d.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
```
Comment (steven-zou) [inline src/common/job/client.go]: maybe better after error checking?

File: src/common/job/client_test.go [hunk 3]
```diff
@@ -47,6 +47,14 @@ func TestGetJobLog(t *testing.T) {
 	assert.Contains(text, "The content in this file is for mocking the get log api.")
 }
 
+func TestGetExecutions(t *testing.T) {
+	assert := assert.New(t)
+	exes, err := testClient.GetExecutions(ID)
+	assert.Nil(err)
+	stat := exes[0]
+	assert.Equal(ID, stat.Info.JobID)
```
Comment (steven-zou) [inline src/common/job/client_test.go]: seems not `==`

File: src/common/job/models/models.go [hunk 4]
```diff
@@ -28,25 +28,28 @@ type JobMetadata struct {
 
 // JobStats keeps the result of job launching.
 type JobStats struct {
```
Comment (steven-zou) [inline src/common/job/models/models.go]: Why NOT directly import job.*?

File: src/core/api/admin_job.go [hunk 5]
```diff
@@ -169,8 +169,22 @@ func (aj *AJAPI) getLog(id int64) {
...
+		if err != nil {
+			aj.SendInternalServerError(err)
+			return
+		}
+		if len(exes) == 0 {
+			aj.SendNotFoundError(errors.New("no execution log "))
+			return
+		}
+		jobID = exes[0].Info.JobID
+	}
```
Comment (steven-zou) [inline src/core/api/admin_job.go]: add else to void the non-necessary case?

File: src/common/job/client.go (hunk 1 above)
Comment (wy65701436) [reply src/common/job/client.go]: will fix, thanks.

File: src/common/job/client_test.go (hunk 3 above)
Comment (wy65701436) [reply src/common/job/client_test.go]: update the mock response

File: src/core/api/admin_job.go (hunk 5 above)
Comment (wy65701436) [reply src/core/api/admin_job.go]: already has

**Pull request merged**